maven-spy/src/main/java/org/jenkinsci/plugins/pipeline/maven/eventspy/handler/DependencyResolutionResultHandler.java

```diff
@@ -42,13 +47,34 @@ protected boolean _handle(DependencyResolutionResult result) {
         Xpp3Dom root = new Xpp3Dom("DependencyResolutionResult");
         root.setAttribute("class", result.getClass().getName());
 
-        /*
         Xpp3Dom dependenciesElt = new Xpp3Dom("resolvedDependencies");
         root.addChild(dependenciesElt);
-        // don't cast to Dependency to not have a classloading issue at runtime
-        for (Object dependency: result.getResolvedDependencies()) {
-            dependenciesElt.addChild(newElement("dependency", dependency.toString()));
-        }*/
+
+        for (Dependency dependency: result.getResolvedDependencies()) {
```

Comment:
Didn't you get classloading issues? Did you try with Maven 3.5.0? I had issues with 3.3.9.

I saw when 3.5.0 was released that they did something with Aether but I did not investigate yet.

If it breaks with Maven <= 3.3.9 then we should have a proper exception handling recommending to use Maven 3.5.0 to benefit of the feature.